### ISSUE TITLE
Result.drop / ResultArtifact.evict — explicit eviction primitive for K-sweep extraction

### ISSUE THREAD

**Issue by AlliedToasters** [AlliedToasters/fpwap#73]
# `Result.drop(layer, hook)` / explicit eviction primitive for K-sweep extraction

## Motivation

`Sweep.run()` returns a `Result` whose `activations(layer, hook)` reads back a memmapped emit. With ragged emit (#65/#66) and PR #71's `as_path=`, the on-disk artifact is correctly self-describing. But the consumer has no first-class way to tell fpwap "I'm done with this layer's emit; you can drop the in-memory references and let the page cache reclaim."

For the K-sweep concat-everything pattern (which paper repro pipelines lean on heavily — see [deception-detection's `_compute_scores_fpwap_concat`](https://github.com/AlliedToasters/deception-detection/blob/feat/fpwap-integration/deception_detection/experiment.py)), this is the difference between "stable host-RAM footprint across K slices" and "linear growth → reclaim cliff at slice ~K/2."

## Symptoms observed

70B repro on RTX 5090 / 128 GB host, CPU residual mode (`fpwap_buffer_device: "cpu"`), K=7 packing the 14k-sample concat, ragged emit on disk via `MemmapBackend`. Run instrumentation watching `/proc/meminfo` + `/proc/PID/io` + `/proc/PID/status` at 1s cadence:

| Sweep | Wall | RSS at end | Cached at end | Notes |
|---|---|---|---|---|
| 1 | 24:20 | 13.3 GB | 5.7 GB | cold weight load |
| 2 | 7:35 | ~22 GB | ~28 GB | accumulating |
| 3 | 4:50 | 23.2 GB | ~43 GB | clean |
| 4 | ~6:00 | 31.0 GB | ~58 GB | clean |
| 5 | **20:51** | 46.3 GB | 66.0 GB | **cliff** |
| 6 | (running, killed) | — | 89.7 GB | — |

At sweep 5 start, committed memory ~89 GB / 128 GB. The next K-slice's residual buffer alloc was ~57 GB (`torch.zeros((1700, 2048, 8192), bf16)` for the seq=2048 bucket). Free RAM was ~24 GB. Linux had to direct-reclaim ~33 GB of file cache + RSS pages while the alloc thread blocked, inflating sweep wall 4x.

Watcher state was mostly `R` during the cliff, not `D` — i.e. CPU-in-kernel-reclaim, not I/O wait. So this is a memory-pressure cliff, distinct from the writeback-throttle pattern previously misdiagnosed in #72.

## Where the retention comes from

Two contributors, both per-sweep:

1. **MemmapBackend emit page cache** — each sweep writes ~14 GB of emits to disk (ragged, post-detection-mask). Those pages stay in `Cached` indefinitely; nothing in fpwap explicitly hints them for eviction after the consumer has read the data.
2. **Consumer-retained Activation slices** — the dd-repo path calls `result.activations(layer, hook)` and slices per-dataset into a cache for later scoring. The slice tensors share storage with the memmap, so each retained reference pins the underlying file's pages. ~3-5 GB per sweep.

Both grow linearly with K-slice count; the first hits the 128 GB ceiling.

## What `as_path=` (PR #71) gives us, and what it doesn't

PR #71 gave the consumer a stable on-disk path. That's load-bearing — it lets a downstream rewrite of `_compute_scores_fpwap_concat` access slices on-demand from disk and explicitly mmap-then-`madvise(DONTNEED)` per slice. Without the stable path, we'd be reaching into MemmapBackend internals.

But PR #71 didn't add the eviction primitive itself. The consumer still has to:

1. Open the path with `os.open` + `mmap.mmap` themselves
2. Build the right tensor view (knowing the bf16-as-u16 layout, the offsets sidecar for ragged, etc.)
3. Call `posix_fadvise(DONTNEED)` on the fd
4. Be careful about ref ordering

Some of that machinery is already inside `MemmapBackend._read_dense_memmap` / `_read_ragged_memmap` (or equivalent helpers). Re-exporting it through a `Result.drop()` (or `ResultArtifact.evict()`) entry point would mean every consumer doesn't reimplement it.

## Proposed API

Two minimal options. I'd favor (B) because it composes with `as_path`:

### (A) `Result.drop(layer, hook)`

```python
result = sweep.run()
acts = result.activations(layer=21, hook="residual_post")
# ... use acts, score, save ...
del acts                          # drops Python ref
result.drop(layer=21, hook="residual_post")
# After this:
#  - any internal refs the Result held to memmap arrays for (21, "residual_post") are released
#  - posix_fadvise(POSIX_FADV_DONTNEED) fired on the underlying emit file fd
#  - subsequent result.activations(layer=21, ...) raises LookupError (or returns None)
```

Idempotent. Safe to call multiple times. Doesn't touch the on-disk file.

### (B) `ResultArtifact.evict()` (hangs off the PR #71 dataclass)

```python
art = result.activations(layer=21, hook="residual_post", as_path=True)
# ... use art.data_path / art.sidecar_path ...
art.evict()  # fadvise(DONTNEED) on art.data_path
```

`evict()` is a no-op if the artifact is already evicted, and is safe even if the underlying file has been hardlinked out (caller's hardlink keeps the inode alive).

Either keeps the eviction policy first-class instead of leaking the kernel-hint plumbing into consumers.

## What the upstream test should assert

A regression test of the form: extract → readback → drop → re-extract → assert `Cached` has not climbed across iterations beyond the in-flight working set. Watcher data from a `/proc/meminfo` poll at 1s cadence is a reasonable harness; the CSV from the run above is available on request.

## Adjacent

- PR [#71](https://github.com/AlliedToasters/fpwap/pull/71) — stable on-disk path; this issue is the eviction half.
- [#50](https://github.com/AlliedToasters/fpwap/issues/50) — fadvise(DONTNEED) on emit shards after write. Inside the sweep loop. This issue is about consumer-side eviction *after readback*.
- [#52](https://github.com/AlliedToasters/fpwap/issues/52) — `MemmapBackend.read` zero-copy. Closed; this one is the dual.
- [#72](https://github.com/AlliedToasters/fpwap/issues/72) — the GPU-allocator-fragmentation issue, separate mechanism. CPU-mode cliff above is *different* (memory reclaim vs CUDA defrag).

Happy to throw together a draft PR if useful.
